subprojects/docs/src/docs/userguide/migration/upgrading_version_8.adoc

```diff
@@ -100,6 +100,47 @@ intended for internal use and was never intended to be used as part of the publi
 
 Starting in Gradle 9.0, this method will be removed without replacement.
 
+[[test_task_default_classpath]]
+==== Relying on Test task default testClassesDirs and classpath
+
+By default, when applying the `JavaPlugin`, standalone `Test` tasks are configured such that their `testClassesDirs`
+and `classpath` mirror the corresponding values in the default `test` `TestSuite`. This behavior will be removed
```

Comment:
I would reword this to repeat what the convention is more explicitly. It doesn't have to be exactly this:

By default, when applying the Java plugin, the `testClassesDirs` and `classpath` for all `Test` tasks have the same convention. Unless changed, a `Test` task will use the tests from the default `test` test suite.